sdk/formrecognizer/ai-form-recognizer/src/formRecognizerClient.ts

```diff
@@ -617,8 +667,149 @@ export class FormRecognizerClient {
     return poller;
   }
 
+  // #endregion
+
+  // #region prebuilt::invoice
+
+  /**
+   * Recognizes data from invoices using a pre-built invoice model, enabling you to extract structured data
+   * from invoices such as customer address, vendor address, purchase order ID, etc.
+   *
+   * For a list of fields that are contained in the response, please refer to the documentation at the following link: https://aka.ms/azsdk/formrecognizer/invoicefields
+   *
+   * This method returns a long running operation poller that allows you to wait
+   * indefinitely until the operation is completed.
+   *
+   * Note that the onProgress callback will not be invoked if the operation completes in the first
+   * request, and attempting to cancel a completed copy will result in an error being thrown.
+   *
+   * Example usage:
+   * ```ts
+   * const path = "./Invoice_1.pdf";
+   * const readStream = fs.createReadStream(path);
+   *
+   * const client = new FormRecognizerClient(endpoint, new AzureKeyCredential(apiKey));
+   * const poller = await client.beginRecognizeInvoices(readStream, {
+   *   contentType: "application/pdf",
+   *   onProgress: (state) => { console.log(`status: ${state.status}`); }
+   * });
+   *
+   * const [invoice] = await poller.pollUntilDone();
+   * ```
+   *
+   * @summary Recognizes invoice information from a given document
+   * @param invoice Input document
+   * @param options Options for the recognition operation
+   */
+  public async beginRecognizeInvoices(
+    invoice: FormRecognizerRequestBody,
+    options: BeginRecognizeInvoicesOptions = {}
+  ): Promise<FormPollerLike> {
+    const { span } = makeSpanner("FormRecognizerClient-beginRecognizeInvoices", {
+      ...options,
+      includeTextDetails: options.includeFieldElements
+    });
+
+    const poller = new FormRecognitionPoller({
+      expectedDocType: "prebuilt:invoice",
+      createOperation: span("invoicesInternal", async (finalOptions) => {
+        const requestBody = await toRequestBody(invoice);
+        const contentType = finalOptions.contentType ?? (await getContentType(requestBody));
+        return processOperationLocation(
+          await this.client.analyzeInvoiceAsync(
+            contentType!,
```

Comment:
I do not like using `!` in the source code of our packages.

Reply:
This might not be needed if the `undefined` case can be removed from `getContentType` now. I'll check on this.

Reply:
Going to take this as part of a refactor in a future version.